tools/test/test_depsolve.py

```diff
@@ -1409,29 +1522,135 @@ def test_depsolve(tmp_path, repo_servers, dnf_config, detect_fn, with_sbom, test
                 assert res["solver"] == "dnf"
 
 
+@pytest.mark.parametrize("with_sbom", [False, True])
+@pytest.mark.parametrize("dnf_config, detect_fn", [
+    ({}, assert_dnf),
+    ({"use_dnf5": False}, assert_dnf),
+    ({"use_dnf5": True}, assert_dnf5),
+], ids=["no-config", "dnf4", "dnf5"])
+def test_depsolve_sbom(repo_servers, dnf_config, detect_fn, with_sbom):
+    try:
+        detect_fn()
+    except RuntimeError as e:
+        pytest.skip(str(e))
+
+    test_case = depsolve_test_case_basic_2pkgs_2repos
+    transactions = test_case["transactions"]
+    tc_repo_servers = get_test_case_repo_servers(test_case, repo_servers)
+    repo_configs = [gen_repo_config(server) for server in tc_repo_servers]
```

Comment:
(nitpick) it seems we have this pattern `    repo_configs = [gen_repo_config(server) for server in tc_repo_servers]` a lot, while its simple enough, maybe make the helper take a list directly like `repo_configs = gen_repo_configs(tc_repo_servers)`

Reply:
I considered this, but since the `config_combos()` needs to do a bit different thing (iterate over specific indexes), and the generation of configs in other places is effectively a single line, I didn't feel like I would be saving much WRT complexity and lines of code. I don't have a strong opinion here; I'm willing to move it to a function.